internal/unmarshal.go

```diff
@@ -167,6 +170,52 @@ func GetValueFromArrowArray(a arrow.Array, idx int) (any, error) {
 	}
 }
 
+type ChunkResult struct {
+	chunkIdx int
+	rows     []map[string]any
+	err      error
+}
+
+func extractFeatures(
+	record arrow.Record,
+	colIndicesShouldSkip []bool,
+	chunkStart int64,
+	chunkEnd int64,
+	chunkIdx int,
+	resChan chan<- ChunkResult,
+	wg *sync.WaitGroup,
+) {
+	defer wg.Done()
+
+	var results []map[string]any
+	chunkEndInt, err := int64ToInt(chunkEnd)
+	if err != nil {
+		resChan <- ChunkResult{chunkIdx: chunkIdx, err: err}
+		return
+	}
+	chunkStartInt := int(chunkStart)
```

Comment:
thank you! but I think since we know chunkStart < chunkEnd and we've already checked chunkEnd is in range, this should be fine. But lmk if you think otherwise